packages/block-editor/src/components/block-navigation/block.js

```diff
@@ -144,7 +157,24 @@ export default function BlockNavigationBlock( {
 								onFocus,
 							} }
 							disableOpenOnArrowDown
-						/>
+							__experimentalSelectBlock={ onClick }
+						>
+							{ ( { onClose } ) => (
+								<MenuGroup>
+									<MenuItem
+										onClick={ async () => {
+											// If clientId is already selected, it won't be focused (see block-wrapper.js)
+											// This removes the selection first to ensure the focus will always switch.
+											await selectEditorBlock( null );
```

Comment:
This seems .... weird. I understand from the comment what it does but the presence of this line feels like something is not happening right.

Reply:
@draganescu I agree, at the same time I don't think it's a blocking issue - I'd love to follow up with a better solution once this PR lands. What do you think?

Reply:
But do we know what that followup will be about?

Reply:
@draganescu the root cause is this effect here:

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-list/block-wrapper.js#L136-L140

As you can see, it applies focus when `isSelected` changes. If the block loses focus but not selection, selecting it again will not trigger that effect. I don't know what a good solution would look like just yet though - I need to give it more thought.

Reply:
Added #23269 to track this!